pytorch_lightning/trainer/connectors/optimizer_connector.py

```diff
@@ -20,7 +20,10 @@ def __init__(self, trainer):
         self.trainer = trainer
 
     def on_trainer_init(self, enable_pl_optimizer):
-        self.trainer._enable_pl_optimizer = enable_pl_optimizer
+        if enable_pl_optimizer is not None:
+            rank_zero_warn(
+                "Trainer argument `enable_pl_optimizer` will be depreacted on 1.3. "
+                "This argument won't be used internally.", DeprecationWarning)
```

Comment:
```suggestion
            rank_zero_warn(
                "Trainer argument `enable_pl_optimizer` is deprecated. It will be removed in v1.3.0",
                DeprecationWarning
            )
```